### DISCUSSION TITLE
Fabric - unwrap_model

### DISCUSSION THREAD

**Discussion by pfeatherstone** [Lightning-AI/pytorch-lightning#18365] [DDP / multi-GPU / multi-node]
I am using Fabric to have a bit more control on the training loop. One of the things it does is export to torchscript and onnx when checkpointing. It looks like i need to convert the `FabricModule` wrapper to a normal `nn.Module`. Is there is an `unwrap_model()` function that takes care of spitting back the original model on cpu, with original fp32 precision while only doing so on rank zero ?
Thank you

**Comment by awaelchli** [comment]
Hey @pfeatherstone 
There is no unwrapping function, but you can always get the original module back by calling

```py
wrapped_model.module
```

**Reply by pfeatherstone** [reply]
Is that the only thing required? HF Accelerate has `accelerator.unwrap_model()`. I assumed it did a bit more than simply return `wrapped_model.module`, like maybe some synchronization between all nodes.

**Reply by pfeatherstone** [reply]
Basically I want to do an ONNX export when checkpointing, which requires unwrapping the model back to CPU
